### PR TITLE
Dolma url count

### PR DESCRIPTION
**Pull request by peterbjorgensen**
Add a simple (single proces) and a complicated (multiprocessing) script to count the domains in a dataset.

### CODE REVIEW

File: data-processing/scripts/dolma_url_count.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from typing import Any, Optional
+from pathlib import Path
+import sys
+
+import smart_open
+import msgspec
+from msgspec import Struct
+import urllib3.util
+
+class DocumentWithMetadata(Struct):
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/dolma_url_count.py:11]: Oh didn't know this was a thing!

**Review by KennethEnevoldsen — APPROVED** [review]
Not much to add here. I assume you have already run it. Is the files located on UCloud?

**Comment by peterbjorgensen** [comment]
The domain list is on ucloud here
`dfm-data/domain-list`

**Comment by KennethEnevoldsen** [comment]
@peterbjorgensen should we have the domain lists on Github (to allow users to remove/add their domains from the list)

**Pull request merged**